### PR TITLE
ARROW-169 Add a benchmark with extension types

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: bindings/python/benchmarks/benchmarks.py [hunk 1]
```diff
@@ -280,6 +281,43 @@ def setup_cache(self):
...
 
 
+class ProfileReadExtensionSmall(Read):
+    schema = Schema({"x": Decimal128Type(), "y": BinaryType(10)})
+    dtypes = np.dtype(np.dtype([("x", np.object_), ("y", np.object_)]))
+
+    def setup_cache(self):
+        coll = db.benchmark
+        coll.drop()
+        base_dict = collections.OrderedDict(
```
Comment (ShaneHarvey) [inline bindings/python/benchmarks/benchmarks.py:291]: Why use OrderedDict?

File: bindings/python/benchmarks/benchmarks.py [hunk 2]
```diff
@@ -280,6 +281,43 @@ def setup_cache(self):
...
+        print(
+            "%d docs, %dk each with %d keys"
+            % (N_DOCS, len(BSON.encode(base_dict)) // 1024, len(base_dict))
+        )
+
+
+class ProfileReadExtensionLarge(Read):
+    large_doc_keys = [f"{i}" for i in range(LARGE_DOC_SIZE)]
+    schema = Schema({k: Decimal128Type() for k in large_doc_keys})
+    dtypes = np.dtype([(k, np.float64) for k in large_doc_keys])
```
Comment (ShaneHarvey) [inline bindings/python/benchmarks/benchmarks.py:307]: How does Decimal128 get converted to np.float64?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/benchmarks/benchmarks.py (hunk 1 above)
Comment (blink1073) [reply bindings/python/benchmarks/benchmarks.py:291]: It was used in the other tests, switched to dict.

File: bindings/python/benchmarks/benchmarks.py (hunk 2 above)
Comment (blink1073) [reply bindings/python/benchmarks/benchmarks.py:307]: Forgot to change it to `object_`, fixed.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**